base/src/iosMain/kotlin/utils/Locale.kt

```diff
@@ -35,38 +35,70 @@ import platform.Foundation.quotationEndDelimiter
 import platform.Foundation.scriptCode
 import platform.Foundation.variantCode
 
-actual class Locale internal constructor(val nsLocale: NSLocale) {
+/**
+ * Default implementation of [BaseLocale]
+ * @property nsLocale the internal [NSLocale] tracking this locale.
+ */
+actual data class Locale internal constructor(val nsLocale: NSLocale) : BaseLocale() {
 
     actual companion object {
+
+        /**
+         * Creates a [Locale] based on a `language` ISO 639 alpha-2 or alpha-3 code
+         * @param language a `language` ISO 639 alpha-2 or alpha-3 code.
+         * @return The [Locale] for the given [language]
+         */
         actual fun createLocale(language: String): Locale = Locale(NSLocale(language))
+
+        /**
+         * Creates a [Locale] based on a ISO 639 alpha-2 or alpha-3 `language` code and ISO 3166 alpha-2 `country` code.
+         * @param language a ISO 639 alpha-2 or alpha-3 `language` code.
+         * @param country a ISO 3166 alpha-2 `country` code.
+         * @return The [Locale] for the given [language] and [country]
+         */
         actual fun createLocale(language: String, country: String): Locale = Locale(NSLocale("${language}_$country"))
+
+        /**
+         * Creates a [Locale] based on a ISO 639 alpha-2 or alpha-3 `language` code, ISO 3166 alpha-2 `country` code, and variant code.
+         * @param language a ISO 639 alpha-2 or alpha-3 `language` code.
+         * @param country a ISO 3166 alpha-2 `country` code.
+         * @param variant Arbitrary value used to indicate a variation of a [Locale]
+         * @return The [Locale] for the given [language], [country], and [variant]
+         */
         actual fun createLocale(language: String, country: String, variant: String): Locale = Locale(NSLocale("${language}_${country}_$variant"))
```

Comment:
When country or variant is empty string (like it defaults below in get()) it will make a string like `en__variant` or `en_US_` which doesn't seems valid. As on Android we actually filter them on non empty and the join with `_`.

We need to do same approach here and above